code/modules/mob/living/simple_animal/abnormality/teth/meat_lantern.dm

```diff
@@ -0,0 +1,149 @@
+	max_boxes = 14
+	ego_list = list(
+		/datum/ego_datum/weapon/lantern,
+		/datum/ego_datum/armor/lantern
+		)
+
+	gift_type = /datum/ego_gifts/lantern
+	gift_message = "Not a single employee has seen Meat Lantern’s full form."
+	var/can_act = TRUE
+	var/detect_range = 1
+	var/chop_cooldown
+	var/chop_cooldown_time = 4 SECONDS
+	var/chop_damage = 400
+
+/mob/living/simple_animal/hostile/abnormality/meat_lantern/PostSpawn()
+	med_hud_set_health() //show medhud while in containment
+	med_hud_set_status()
+
+/mob/living/simple_animal/hostile/abnormality/meat_lantern/PickTarget(list/Targets)
+	return FALSE
+
+/mob/living/simple_animal/hostile/abnormality/meat_lantern/AttackingTarget()
+	return FALSE
+
+/mob/living/simple_animal/hostile/abnormality/meat_lantern/Move()
+	return FALSE
+
+/mob/living/simple_animal/hostile/abnormality/meat_lantern/FearEffect() //makes it too easy to find with a giant exclamation mark over your head
+	return
+
+/mob/living/simple_animal/hostile/abnormality/meat_lantern/death()
+	. = ..()
+	gib()
+
+/mob/living/simple_animal/hostile/abnormality/meat_lantern/HasProximity(atom/movable/AM)
+	if(!isliving(AM))
+		return
+	var/mob/living/L = AM
+	if(L.stat == DEAD)
+		return
+	if(!can_act || (chop_cooldown > world.time))
+		return
+	INVOKE_ASYNC(src, .proc/BigChop)
+
+/mob/living/simple_animal/hostile/abnormality/meat_lantern/proc/BigChop()
+	can_act = FALSE
+	new /obj/effect/temp_visual/yellowsmoke(get_turf(src))
+	icon = 'ModularTegustation/Teguicons/224x128.dmi'
+	flick("lantern_bite_open",src)
+	pixel_x = base_pixel_x - 88
+	playsound(get_turf(src), 'sound/effects/ordeals/amber/midnight_out.ogg', 40)
+	SLEEP_CHECK_DEATH(7)
+	icon = 'ModularTegustation/Teguicons/128x128.dmi'
+	flick("lantern_bite_closed", src)
+	pixel_x = base_pixel_x - 40
+	for(var/mob/living/L in oview(1, src))
+		if(faction_check_mob(L))
+			return
+		L.apply_damage(chop_damage, RED_DAMAGE, null, L.run_armor_check(null, RED_DAMAGE), spread_damage = TRUE)
+		if(L.health < 0)
+			L.gib(FALSE,FALSE,TRUE)
+	SLEEP_CHECK_DEATH(2.5)
+	icon = initial(icon)
+	pixel_x = base_pixel_x
+	can_act = TRUE
+	chop_cooldown = world.time + chop_cooldown_time
+	addtimer(CALLBACK(src, .proc/ProximityCheck), chop_cooldown_time)
+
+/mob/living/simple_animal/hostile/abnormality/meat_lantern/proc/ProximityCheck()
+	for(var/mob/living/L in range(1,src)) //hidden istype() call
+		if(L == src)
+			continue
+		BigChop()
+		return
+
+/mob/living/simple_animal/hostile/abnormality/meat_lantern/med_hud_set_health()
+	if(!IsContained())
+		var/image/holder = hud_list[HEALTH_HUD]
+		holder.icon_state = null
+		return
+	..()
+
+/mob/living/simple_animal/hostile/abnormality/meat_lantern/med_hud_set_status()
+	if(!IsContained())
+		var/image/holder = hud_list[STATUS_HUD]
+		holder.icon_state = null
+		return
+	..()
+
+/mob/living/simple_animal/hostile/abnormality/meat_lantern/PostWorkEffect(mob/living/carbon/human/user, work_type, pe, work_time, canceled)
+	if(work_time < 19 SECONDS)
```

Comment:
I would prefer this being a temperance check. It's a lot harder to be sure of work times and will require external tools.
Talk to redacted about it

Reply:
Work Time check works against both people with stronger temperance and those who cancel work. 2 birds with one stone honestly.